test/blackbox-tests/test-cases/pkg/lockdir-validation.t

```diff
@@ -50,3 +50,17 @@ package graph then it's caught when loading the lockdir.
   Hint: This could indicate that the lockdir is corrupted. Delete it and then
   regenerate it by running: 'dune pkg lock'
   [1]
+
+  $ cat >dune.lock/a.pkg <<EOF
+  > (deps b b)
+  > EOF
+
+  $ dune describe pkg lock
+  File "dune.lock/a.pkg", line 1, characters 8-9:
+  1 | (deps b b)
+              ^
+  Error: Package "a" in lockdir dune.lock contains duplicate dependencies on
+  package "b".
+  Hint: This could indicate that the lockdir is corrupted. Delete it and then
+  regenerate it by running: 'dune pkg lock'
```

Comment:
Mostly I'm interested in detecting user modifications to the lockdir that cause it to no longer contain a solution to local package dependencies. I agree that this check is possibly too strict. I added it because we keep track of the `Loc.t` of each dependency in the lockdir and I didn't want to think about how to handle the case where a dependency appears twice. Also if a dependency appears twice then the lockfile won't round-trip after this change. Neither of those are big deals though.

Reply:
Yeah, I agree, after I wrote the comment I saw the code that computes it and indeed the validation comes sort of "for free".

I don't know, what do the others think?

Reply:
To preserve the order of dependencies I'll be changing the representation of lockfile dependencies back to a list so there's no structural reason to not allow duplicates. I'll get rid of this check for now.